x/wasm/keeper/keeper.go

```diff
@@ -655,6 +655,8 @@ func (k Keeper) QuerySmart(ctx sdk.Context, contractAddr sdk.AccAddress, req []b
 	queryResult, gasUsed, qErr := k.wasmVM.Query(codeInfo.CodeHash, env, req, prefixStore, cosmwasmAPI, querier, k.gasMeter(ctx), k.runtimeGasForContract(ctx), costJSONDeserialization)
 	k.consumeRuntimeGas(ctx, gasUsed)
 	if qErr != nil {
+		// consume ALL remaining gas on error
+		// k.consumeRemainingGas(ctx)
```

Comment:
is this commented out intentionally?

Reply:
ah no, I was commenting it out to intentionally verify behavior with and without the change, good catch.